mlflow/tracing/fluent.py

```diff
@@ -549,3 +550,172 @@ def f():
+    service to the current active trace in the function.
+
+    .. code-block:: python
+
+        @mlflow.trace(name="predict")
+        def predict(input):
+            # Call a remote service that returns a trace in the response
+            resp = requests.get("https://your-service-endpoint", ...)
+
+            # Extract the trace from the response
+            trace_json = resp.json().get("trace")
+
+            # Use the remote trace as a part of the current active trace.
+            # It will be merged under the span "predict" and exported together when it is ended.
+            mlflow.add_trace(trace_json)
+
+    If you have a specific target span to merge the trace under, you can pass the target span
+
+    .. code-block:: python
+
+        def predict(input):
+            # Create a local span
+            span = MlflowClient().start_span(name="predict")
+
+            resp = requests.get("https://your-service-endpoint", ...)
+            trace_json = resp.json().get("trace")
+
+            # Merge the remote trace under the span created above
+            mlflow.add_trace(trace_json, target=span)
+
+    Args:
+        trace: A :py:class:`Trace <mlflow.entities.Trace>` object or a dictionary representation
+            of the trace. The trace **must** be already completed i.e. no further updates should
+            be made to it. Otherwise, this function will raise an exception.
+
+            .. attention:
+
+                The spans in the trace must be ordered in a way that the parent span comes
+                before its children. If the spans are not ordered correctly, this function
+                will raise an exception.
+
+        target: The target span to merge the given trace.
+
+            - If provided, the trace will be merged under the target span.
+            - If not provided, the trace will be merged under the current active span.
+            - If not provided and there is no active span, a new span named "Remote Trace <...>"
+                will be created and the trace will be merged under it.
+    """
+    if isinstance(trace, dict):
+        try:
+            trace = Trace.from_dict(trace)
+        except Exception as e:
+            raise MlflowException.invalid_parameter_value(
+                "Failed to load a trace object from the given dictionary. Please ensure the "
+                f"dictionary is in the correct MLflow Trace format. Error: {e}",
+            )
+    elif not isinstance(trace, Trace):
+        raise MlflowException.invalid_parameter_value(
+            f"Invalid trace object: {type(trace)}. Please provide a valid MLflow Trace object "
+            "to use it as a remote trace. You can create a Trace object from its json format by "
+            "using the Trace.from_dict() method."
+        )
+
+    if trace.info.status not in TraceStatus.end_statuses():
+        raise MlflowException.invalid_parameter_value("The remote trace must be ended.")
```

Comment:
can we include `trace.info.status` in the message?

Reply:
Good idea!